modules/katello_composite_content_view.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2017, Greg Swift <gregswift@gmail.com>
+# Copied from katello_content_view.py by Eric Helms
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: katello_composite_content_view
+short_description: Create and Manage Katello composite content views
+description:
+    - Create and Manage Katello composite content views
+author: "Greg Swift (@gregswift)"
+requirements:
+    - "nailgun >= 0.28.0"
+    - "python >= 2.6"
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        required: false
+        default: true
+        type: bool
+    name:
+        description:
+            - Name of the Katello composite content view
+        required: true
+    organization:
+        description:
+            - Organization that the Content Views are in
+        required: true
+    content_views:
+        description:
+            - List of content views that include name and version
+        required: false
+        type: list
+    state:
+        description:
+        - State of the Global Parameter
+        required: true
+        choices:
+        - present
+        - absent
+        - present_with_defaults
+'''
+
+EXAMPLES = '''
+- name: "Create or update a Fedora composite content view"
+  katello_composite_content_view:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "OurFedora CCV"
+    organization: "My Cool new Organization"
+    content_views:
+      - name: 'Fedora CV'
+        version: '2.0'
+      - name: 'Ansible CV'
+        version: '1.0'
```

Comment:
I'd add `state` to this example since it is required.

Reply:
Or should I change state to `default: true`

Reply:
It would be either 'present' or 'absent', and i think following the other modules would be better.

Reply:
generally speaking i think ansible defaults this to true and doesn't require it as a parameter (even if putting it there is considered a best practice)

Reply:
I'd like to have this consistent across all entity modules. So we might discuss that in a general issue.
And again, it were `default: 'present'`, not `true`.

Reply:
#104 Present is the default now.